File: BookstoreApp/app/src/main/java/com/example/bookstoreapp/RegistrationScreen.java
package com.example.bookstoreapp;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class RegistrationScreen extends AppCompatActivity {

    EditText etName, etEmail, etPassword, etConfirmPassword;
    Button btnRegister;
    DatabaseHelper db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.screen_register);

        // Initialize views
        etName = findViewById(R.id.et_name);
        etEmail = findViewById(R.id.et_email);
        etPassword = findViewById(R.id.et_password);
        etConfirmPassword = findViewById(R.id.et_confirm_password); // Add confirm password field
        btnRegister = findViewById(R.id.btn_register);
        db = new DatabaseHelper(this);

        // Set button click listener
        btnRegister.setOnClickListener(view -> {
            String name = etName.getText().toString().trim();
            String email = etEmail.getText().toString().trim();
            String password = etPassword.getText().toString().trim();
            String confirmPassword = etConfirmPassword.getText().toString().trim();

            // Validate inputs
            if (name.isEmpty() || email.isEmpty() || password.isEmpty() || confirmPassword.isEmpty()) {
                Toast.makeText(RegistrationScreen.this, "Please fill in all fields", Toast.LENGTH_SHORT).show();
                return;
            }

            // Check if passwords match
            if (!password.equals(confirmPassword)) {
                Toast.makeText(RegistrationScreen.this, "Passwords do not match!", Toast.LENGTH_SHORT).show();
                return;
            }

            // Register user in the database
            if (db.registerUser(name, email, password)) {
                Toast.makeText(RegistrationScreen.this, "Registration successful!", Toast.LENGTH_SHORT).show();
                // Navigate to the login or home screen
                startActivity(new Intent(RegistrationScreen.this, LoginScreen.class));
                finish();
            } else {
                Toast.makeText(RegistrationScreen.this, "Registration failed. Try again.", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: BikeCourierKotlin/app/src/main/java/com/application/bikecourier_kotlin/RegistrationScreen.kt
package com.application.bikecourier_kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class RegistrationScreen : AppCompatActivity() {
    private lateinit var dbHelper: DatabaseHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_registrationscreen)

        dbHelper = DatabaseHelper(this)

        val etPhoneNumber: EditText = findViewById(R.id.etRegisterPhoneNumber)
        val etName: EditText = findViewById(R.id.etRegisterName)
        val etPassword: EditText = findViewById(R.id.etRegisterPassword)
        val etConfirmPassword: EditText = findViewById(R.id.etConfirmPassword)
        val btnRegister: Button = findViewById(R.id.btnRegister)
        val tvSignIn: TextView = findViewById(R.id.tvSignIn)

        btnRegister.setOnClickListener {
            val phone = etPhoneNumber.text.toString().trim()
            val name = etName.text.toString().trim()
            val password = etPassword.text.toString().trim()
            val confirmPassword = etConfirmPassword.text.toString().trim()

            if (phone.isEmpty() || name.isEmpty() || password.isEmpty() || confirmPassword.isEmpty()) {
                Toast.makeText(this, "Please fill all fields", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (password != confirmPassword) {
                Toast.makeText(this, "Passwords do not match", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (dbHelper.registerUser(phone, name, password)) {
                Toast.makeText(this, "Registration successful!", Toast.LENGTH_SHORT).show()
                startActivity(Intent(this, LoginScreen::class.java))
                finish()
            } else {
                Toast.makeText(this, "Registration failed! Phone number already exists.", Toast.LENGTH_SHORT).show()
            }
        }

        tvSignIn.setOnClickListener {
            startActivity(Intent(this, LoginScreen::class.java))
            finish()
        }
    }
}